pkg/document/crdt/tree.go

```diff
@@ -876,6 +876,34 @@ func (t *Tree) Style(from, to *TreePos, attributes map[string]string, editedAt *
 	return nil
 }
 
+// RemoveStyle removes the given attributes of the given range.
+func (t *Tree) RemoveStyle(from, to *TreePos, attributesToRemove []string, editedAt *time.Ticket) error {
+	// 01. split text nodes at the given range if needed.
+	fromParent, fromLeft, err := t.FindTreeNodesWithSplitText(from, editedAt)
+	if err != nil {
+		return err
+	}
+	toParent, toLeft, err := t.FindTreeNodesWithSplitText(to, editedAt)
+	if err != nil {
+		return err
+	}
+
+	err = t.traverseInPosRange(fromParent, fromLeft, toParent, toLeft,
+		func(token index.TreeToken[*TreeNode], _ bool) {
+			node := token.Node
+			if !node.IsRemoved() && !node.IsText() && len(attributesToRemove) > 0 && node.Attrs != nil {
```

Comment:
How about removing `len(attributesToRemove) > 0 && node.Attrs != nil` conditions in this line? I think empty `attributesToRemove` should not be passed here and `node.Attrs` should be created before calling this method.

```suggestion
			if !node.IsRemoved() && !node.IsText() {
```

Reply:
The condition `len(attributesToRemove) > 0` can be removed, but `node.Attrs != nil` is necessary because a user might call `Tree.RemoveStyle` without having called `Tree.Style` previously.

Reply:
I modified the code to create a new `RHT` if `node.Attrs` is `nil` to support concurrency.